frontends/web/src/routes/buy/exchange.tsx

```diff
@@ -178,14 +182,13 @@ export const Exchange = ({ code, accounts }: TProps) => {
   const bankTransferFee = infoFeesDetail && infoFeesDetail.find(feeDetail => feeDetail.payment === 'bank-transfer')?.fee;
 
   const handleGoBack = () => {
-    if (accounts.length > 1) {
-      return navigate(-1);
+    if (supportedAccounts.length > 1) {
+      // go back to the account selection page
+      route('/buy/info');
+    } else {
+      // go back to home screen
+      route('/account-summary');
```

Comment:
might be slightly confusing if the user didn't come from `/account-summary`. 

E.g they might come from `settings` (before going into `/buy/info` which auto-redirects them to this page `/buy/exchange/accountId`).

Clicking on "back", they most likely would expect to go back to `settings` and not `account-summary`.  Just a nit ig.. 🤔 🤷

Reply:
Hm... 

I think the bit of code that you changed would do what I suggested.

```js
if (accounts.length > 1) {
      return navigate(-1);
```

Will go back to `/buy/info`

... otherwise

```js
//otherwise
navigate(-2);
```

will go back to `/buy/info` then go back to whatever page before that, which is the page that the user went before clicking on "Buy Crypto / Bitcoin".

But... if `navigate` doesn't work on Android then yeah, have to find an alternative.

Reply:
A little strange though that it doesn't work on Android. Did you find any bug report or people facing the same issue on android, I guess chromium (mobile ?)

Reply:
Good point. I could not find anything about this issue online, and I am not sure how to log that behavior on Android.
Could we try a different library maybe? Like React Navigation? https://reactnavigation.org/

Reply:
> Could we try a different library maybe? Like React Navigation? https://reactnavigation.org/

Might be a bit overkill if it's only for this purpose.. 🤔 

> We could also think about removing the Back button if there is only one account with available exchanges.

I initially agree w/ this, but thinking back.. could be annoying for users (especially mobile) if there's no back btn. To go out from the `/buy/exchange` user needs to click on the hamburger menu, then clicking on the desired menu. 

Especially not a very good experience since the built-in back btn on Android isn't working  if I'm not mistaken.

Let's hear thisconnect's opinion on this.

Reply:
> I initially agree w/ this, but thinking back.. could be annoying for users (especially mobile) if there's no back btn. To go out from the /buy/exchange user needs to click on the hamburger menu, then clicking on the desired menu.

Isn't the same in every other section of the app? I think we usually use the back button only in flows and situations where there is a clear parent page. Anyway yes, let's hear @thisconnect thoughts about this.

Reply:
@thisconnect we briefly discussed this yesterday, but I don't think we actually took a decision. wdyt?

Reply:
browser back would have been an relatively simple solution, but as it doesnt' work on Android (assuming this is a limitation of android webview) I think we should only show the backbutton if it goes back to account selection. If there is only 1 account in the selection we can hide the back button for now. 

account selection doesn't have back button. account selection backbutton would go to different views, depending on where the user came from.

maybe one day we can look into a general solution for back (also android native back button) and look into why  history back doesn't work on anrdoid.